platform/security/filter/security-filter-authorization/src/main/java/org/codice/ddf/security/filter/authorization/AuthorizationFilter.java

```diff
@@ -94,19 +94,16 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 if (StringUtils.isEmpty(path)) {
                     path = httpRequest.getRequestURI();
                 }
-                ActionPermission actionPermission = new ActionPermission(path);
+
                 ContextPolicy policy = contextPolicyManager.getContextPolicy(path);
 
-                if (policy != null) {
+                if (policy != null && subject != null) {
                     Collection<CollectionPermission> permissions = policy
                             .getAllowedAttributePermissions();
-                    if (!CollectionUtils.isEmpty(permissions)) {
-                        permissions.add(new CollectionPermission(actionPermission));
-                        for (CollectionPermission permission : permissions) {
-                            if (subject == null || !subject
-                                    .isPermittedAll(permission.getPermissionList())) {
-                                permitted = false;
-                            }
+                    for (CollectionPermission permission : permissions) {
```

Comment:
The action is already build into the returned CollectionPermissions from when they were created. I updated the ContextPolicy and Policy classes to return a single CollectionPermission. The AuthorizationFilter now just calls isPermitted once with the CollectionPermission it gets from the context policy.

Also updated DefaultContextAttributeMapping for single KeyValuePermission instead of a CollectionPermission